io/net/bonding.py

```diff
@@ -286,16 +326,17 @@ def tearDown(self):
             if process.system(cmd, shell=True, ignore_status=True) != 0:
                 self.log.info("unable to bring up to original state in host")
             time.sleep(5)
-        self.bond_remove("peer")
-        for val1, val2, val3 in map(None, self.peer_interfaces,
-                                    self.peer_ips, self.net_mask):
-            msg = "ifconfig %s %s netmask %s up;"\
-                  % (val1, val2, val3)
-            cmd = "ssh %s@%s \"%s\""\
-                  % (self.user, self.peer_first_ipinterface, msg)
-            if process.system(cmd, shell=True, ignore_status=True) != 0:
-                self.log.info("unable to bring up to original state in host")
-            time.sleep(5)
+        if self.peer_bond_needed:
+            self.bond_remove("peer")
+            for val1, val2, val3 in map(None, self.peer_interfaces,
+                                        self.peer_ips, self.net_mask):
+                msg = "ifconfig %s %s netmask %s up;"\
+                      % (val1, val2, val3)
+                cmd = "ssh %s@%s \"%s\""\
+                      % (self.user, self.peer_first_ipinterface, msg)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring to original state in host")
+                time.sleep(5)
 
```

Comment:
Can you please submit job.log, if possible with peer_bond_needed value as True. As good amount of code executes with that condition in your script.